authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,998 @@
+  }
+
+  @Override
+  public Boolean onRevokedRolesFromGroup(List<Role> roles, Group group) throws RuntimeException {
+    onGroupAdded(group);
+    roles.stream()
+        .forEach(
+            role -> {
+              checkRangerRole(role.name());
+              GrantRevokeRoleRequest grantRevokeRoleRequest =
+                  createGrantRevokeRoleRequest(role.name(), null, group.name());
+              try {
+                rangerClient.revokeRole(rangerServiceName, grantRevokeRoleRequest);
+              } catch (RangerServiceException e) {
+                // Ignore exception to support idempotent operation
+                LOG.warn("Revoke role: {} from group: {} failed!", role, group, e);
+              }
+            });
+
+    return Boolean.TRUE;
+  }
+
+  @Override
+  public Boolean onUserAdded(User user) throws RuntimeException {
+    VXUserList list = rangerClient.searchUser(ImmutableMap.of("name", user.name()));
+    if (list.getListSize() > 0) {
+      LOG.warn("The user({}) is already exist in the Ranger!", user.name());
+      return Boolean.FALSE;
+    }
+
+    VXUser rangerUser = VXUser.builder().withName(user.name()).withDescription(user.name()).build();
+    return rangerClient.createUser(rangerUser);
+  }
+
+  @Override
+  public Boolean onUserRemoved(User user) throws RuntimeException {
+    VXUserList list = rangerClient.searchUser(ImmutableMap.of("name", user.name()));
+    if (list.getListSize() == 0) {
+      LOG.warn("The user({}) is not exist in the Ranger!", user);
+      return Boolean.FALSE;
+    }
+    rangerClient.deleteUser(list.getList().get(0).getId());
+    return Boolean.TRUE;
+  }
+
+  @Override
+  public Boolean onUserAcquired(User user) throws RuntimeException {
+    VXUserList list = rangerClient.searchUser(ImmutableMap.of("name", user.name()));
+    if (list.getListSize() == 0) {
+      LOG.warn("The user({}) is not exist in the Ranger!", user);
+      return Boolean.FALSE;
+    }
+    return Boolean.TRUE;
+  }
+
+  @Override
+  public Boolean onGroupAdded(Group group) throws RuntimeException {
+    return rangerClient.createGroup(VXGroup.builder().withName(group.name()).build());
+  }
+
+  @Override
+  public Boolean onGroupRemoved(Group group) throws RuntimeException {
+    VXGroupList list = rangerClient.searchGroup(ImmutableMap.of("name", group.name()));
+    if (list.getListSize() == 0) {
+      LOG.warn("The group({}) is not exists in the Ranger!", group);
+      return Boolean.FALSE;
+    }
+    return rangerClient.deleteGroup(list.getList().get(0).getId());
+  }
+
+  @Override
+  public Boolean onGroupAcquired(Group group) {
+    VXGroupList vxGroupList = rangerClient.searchGroup(ImmutableMap.of("name", group.name()));
+    if (vxGroupList.getListSize() == 0) {
+      LOG.warn("The group({}) is not exists in the Ranger!", group);
+      return Boolean.FALSE;
+    }
+    return Boolean.TRUE;
+  }
+
+  private boolean checkRangerRole(String roleName) throws AuthorizationPluginException {
```

Comment:
Maybe this method should use void as return value.

Reply:
DONE